tests/test_lora.py

```diff
@@ -422,7 +422,15 @@ def test_lora_merge_with_quantize():
 
     # check that `W_after = W_initial + (A x B)`
     delta_w = (attn_proj.lora_B @ attn_proj.lora_A) * attn_proj.scaling
-    torch.testing.assert_close(weight_after, initial_weight + delta_w)
+    # dequantize initial weight and sum with delta_w
+    initial_weight_data = (
+        bnb.functional.dequantize_4bit(initial_weight.data, initial_weight_kwargs["quant_state"]) + delta_w
+    )
+    # quantize again
+    initial_weight_data = bnb.nn.Params4bit(
+        initial_weight_data.to("cpu"), requires_grad=False, **initial_weight_kwargs
+    ).to(initial_weight.device)
```

Comment:
Can you create a function that does this and call it from both the test and the source code?

Reply:
Yes, I can.
I'll create something like this after I settle on the solution: dequantize, sum and quantize or, like you proposed, quantize and sum.